### PR TITLE
feat: Add Github Action for notation sign with plugin

### PR DESCRIPTION
**Pull request by Two-Hearts**
The Github Action is tested with this build: https://github.com/notation-playground/notation-integration-with-ACR-and-AKV/actions/runs/5487677806/jobs/9999457179

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,25 @@
...
+    - name: sign releasd artifact with signing plugin
+      uses: notaryproject/notation-action/sign@main
+      with:
+        plugin_name: <notation_signing_plugin_name>
+        plugin_url: <plugin_download_url>
+        key_id: <key_identifier_to_sign>
+        target_artifact_reference: <target_artifact_reference_in_remote_registry>
+        plugin_config: <plugin_defined_config>
+    ```
+    For example,
```
Comment (FeynmanZhou) [inline README.md:15]: @priteshbandi @iamsamirzon Would you want to add the AWS Signer plugin as an example?

File: sign/action.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: notation-sign
+description: notation sign with plugin
+inputs:
+  plugin_name:
+    description: name of the signing plugin
+    required: true
+  plugin_url:
+    description: download url of the signing plugin
+    required: true
```
Comment (shizhMSFT) [inline sign/action.yml:9]: When downloading from a plugin URL, the corresponding checksum must be provided.

File: sign/index.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        } else {
+            core.setFailed('Unknown error');
+        }
+    }
+}
+
+export = sign;
+
+if (require.main === module) {
+  sign();
```
Comment (shizhMSFT) [inline sign/index.ts:80]: nit: indentation?

File: sign/index.ts [hunk 4]
```diff
@@ -42,15 +43,21 @@ async function sign() {
...
         const plugin_url = core.getInput('plugin_url');
+        const plugin_checksum = core.getInput('plugin_checksum');
         console.log(`signing plugin url is ${plugin_url}`);
         const pluginPath = os.homedir() + `/.config/notation/plugins/${plugin_name}`;
         fs.mkdirSync(pluginPath, { recursive: true, });
 
         const pathToTarball = await tc.downloadTool(plugin_url);
+        const buff = fs.readFileSync(pathToTarball);
+        const sha256 = hash(buff);
+        if (sha256 !== plugin_checksum) {
```
[the remark this answers is not in the document]
Comment (Two-Hearts) [reply sign/index.ts:58]: Nice catch!

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: sign/index.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    }
+}
+
+// setup the signing Sign.
+async function setupPlugin() {
+    try {
+        const plugin_url = core.getInput('plugin_url');
+        const plugin_checksum = core.getInput('plugin_checksum').toLowerCase();
+        console.log(`signing plugin url is ${plugin_url}`);
+        const pluginPath = os.homedir() + `/.config/notation/plugins/${plugin_name}`;
```
Comment (JeyJeyGao) [inline sign/index.ts:49]:
```typescript
function getConfigHome(): string {
    const platform = os.platform();
    switch (platform) {
        case 'win32':
            if (process.env.APPDATA === undefined) {
                throw new Error('APPDATA is undefined');
            }
            return process.env.APPDATA;
        case 'darwin':
            if (process.env.HOME === undefined) {
                throw new Error('HOME is undefined');
            }
            return path.join(process.env.HOME, 'Library', 'Application Support');
        case 'linux':
            if (process.env.XDG_CONFIG_HOME === undefined) {
                throw new Error('XDG_CONFIG_HOME is undefined');
            }
            return process.env.XDG_CONFIG_HOME;
        default:
            throw new Error(`Unknown platform: ${platform}`);
    }
}
```
We may need to get the config home directory per platform.

File: sign/index.ts [hunk 6]
```diff
@@ -13,27 +13,23 @@ const plugin_name = core.getInput('plugin_name');
...
+                await exec.getExecOutput(`notation sign --signature-format ${signature_format} --allow-referrers-api --id ${key_id} --plugin ${plugin_name} ${target_artifact_ref}`);
             }
-            console.log('notation sign output:\n', signOutput);
         } else {
             if (plugin_config) {
-                signOutput = execSync(`notation sign --signature-format ${signature_format} --id ${key_id} --plugin ${plugin_name} --plugin-config=${plugin_config} ${target_artifact_ref}`, { encoding: 'utf-8' });
+                await exec.getExecOutput(`notation sign --signature-format ${signature_format} --id ${key_id} --plugin ${plugin_name} --plugin-config=${plugin_config} ${target_artifact_ref}`);
             } else {
-                signOutput = execSync(`notation sign --signature-format ${signature_format} --id ${key_id} --plugin ${plugin_name} ${target_artifact_ref}`, { encoding: 'utf-8' });
+                await exec.getExecOutput(`notation sign --signature-format ${signature_format} --id ${key_id} --plugin ${plugin_name} ${target_artifact_ref}`);
```
Comment (shizhMSFT) [inline sign/index.ts]: Can we leave the first parameter just be `notation`? and leave others to the `args`?

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Pull request merged**